src/app/components/mnoe-support/support-organizations-list.coffee

```diff
@@ -66,22 +73,46 @@
         ]
       )
 
-    scope.searchChange = () ->
-      scope.searchMode = true
-      setSearchOrganizationsList(scope.organizations.search)
+    scope.externalIdSearch = () ->
+      return unless scope.organizations.externalIdSearch.length > 2
+      # Reset other search fields.
+      scope.organizations.orgNameSearch = ''
+      scope.organizations.firstNameSearch = ''
+      scope.organizations.lastNameSearch = ''
+      setSearchOrganizationsList()
+
+    scope.nameSearch = () ->
+      org = scope.organizations
+      return unless org.orgNameSearch.length >= 2 && org.firstNameSearch.length >= 2 && org.lastNameSearch.length >= 2
```

Comment:
Does it need to be all three conditions? Or just one of them?

Reply:
@adamaziz15 yep all three due to requirements.